chimeraTKApp/src/ChimeraTK/EPICS/ArrayRecordDeviceSupport.h

```diff
@@ -116,11 +116,14 @@ class ArrayRecordDeviceSupportTrait : public RecordDeviceSupportBase {
   /**
    * Updates the record's TIME field with the specified time stamp.
    */
-  void updateTimeStamp(TimeStamp const &timeStamp) {
+  void updateTimeStamp(VersionNumber const &versionNumber) {
+    auto time = versionNumber.getTime();
+    double timeInNanosecs = std::chrono::time_point_cast<std::chrono::nanoseconds>(time).time_since_epoch().count();
```

Comment:
I do not like the use of a `double` here. The mantissa of a double is only 52 bits, which means that effectively, we will not have nanosecond precision, but rather something like microsecond precision (depending on the exact time stamp, of course).

Why do we not use `std::int64_t` here or - if you prefer that - `std::chrono::nanoseconds::rep`?

Reply:
You are right, I will change it to std::chrono::nanoseconds::rep (which is int64_t usually, I think).